src/runtime-rs/crates/runtimes/src/tracer.rs

```diff
@@ -0,0 +1,164 @@
+// Copyright (c) 2019-2022 Alibaba Cloud
+// Copyright (c) 2019-2022 Ant Group
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+use std::cmp::min;
+use std::sync::Arc;
+use std::sync::Mutex;
+
+use anyhow::Result;
+use lazy_static::lazy_static;
+use opentelemetry::global;
+use opentelemetry::runtime::Tokio;
+use tracing::{span, subscriber::NoSubscriber, Span, Subscriber};
+use tracing_subscriber::prelude::*;
+use tracing_subscriber::Registry;
+
+lazy_static! {
+    /// The static KATATRACER is the variable to provide public functions to outside modules
+    static ref KATATRACER: Arc<Mutex<KataTracer>> = Arc::new(Mutex::new(KataTracer::new()));
```

Comment:
I am wondering if it is possible to maintain the global tracer life cycle through the manager without using the global variable KATATRACER

Reply:
Currently I haven't think of a way to get this done, since when entering and exiting both needs some information (now only enabled), but I wrap this to a struct for extensibility.